### PR TITLE
[refactor] Make sure we have the expected typing in visitors

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Refs https://github.com/PyCQA/pylint/pull/6636#discussion_r891485685

### CODE REVIEW

**Review by cdce8p — CHANGES_REQUESTED** [review]
I would recommend to limit this refactoring to the suggestion in https://github.com/PyCQA/pylint/pull/6636#discussion_r891485685. All other changes aren't worth it IMO.

**Comment by Pierre-Sassoulas** [comment]
I checked for all the call of a visitor function directly (``self.visit_``) and added the typing for them.

**Review by cdce8p — APPROVED** [review]
Looks good. One last comment, can be merged after that

File: pylint/extensions/broad_try_clause.py [hunk 1]
```diff
@@ -58,13 +58,15 @@ def _count_statements(self, try_node: nodes.TryExcept | nodes.TryFinally) -> int
     def visit_tryexcept(self, node: nodes.TryExcept | nodes.TryFinally) -> None:
         try_clause_statements = self._count_statements(node)
         if try_clause_statements > self.linter.config.max_try_statements:
-            msg = f"try clause contains {try_clause_statements} statements, expected at most {self.linter.config.max_try_statements}"
+            msg = (
+                f"try clause contains {try_clause_statements} statements, expected at"
+                f" most {self.linter.config.max_try_statements}."
```
Comment (cdce8p) [inline pylint/extensions/broad_try_clause.py:63]:
```suggestion
                f" most {self.linter.config.max_try_statements}"
```
The trailing `.` is causing the CI failure.

File: pylint/extensions/docparams.py [hunk 2]
```diff
@@ -341,7 +341,7 @@ def visit_return(self, node: nodes.Return) -> None:
         if not (doc.has_rtype() or (doc.has_property_type() and is_property)):
             self.add_message("missing-return-type-doc", node=func_node)
 
-    def visit_yield(self, node: nodes.Yield) -> None:
+    def visit_yield(self, node: nodes.Yield | nodes.YieldFrom) -> None:
```
[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply pylint/extensions/docparams.py:344]: Nice catch !

**Pull request merged**